packages/flutter_tools/lib/src/build_system/file_hash_store.dart

```diff
@@ -143,24 +135,28 @@ class FileHashStore {
 
   /// Computes a hash of the provided files and returns a list of entities
   /// that were dirty.
-  // TODO(jonahwilliams): compare hash performance with md5 tool on macOS and
-  // linux and certutil on Windows, as well as dividing up computation across
-  // isolates. This also related to the current performance issue with checking
-  // APKs before installing them on device.
   Future<List<File>> hashFiles(List<File> files) async {
     final List<File> dirty = <File>[];
-    for (File file in files) {
-      final String absolutePath = file.resolveSymbolicLinksSync();
-      final String previousHash = previousHashes[absolutePath];
-      final List<int> bytes = file.readAsBytesSync();
-      final String currentHash = md5.convert(bytes).toString();
+    final Pool openFiles = Pool(kMaxOpenFiles);
```

Comment:
The fix was moving this up here instead of declaring it inline. That had caused one to be created for each call to `_hashFile`, making it ineffective at limiting the number of open files

Reply:
See comment below. I'm wondering if one global pool with a bigger limit is the way to go.